sdk/metric/sdk.go

```diff
@@ -139,6 +143,15 @@ type (
 		labels        *label.Set
 		observed      export.Aggregator
 	}
+
+	// MetricsProcessor can be provided as an config option when creating
+	// a new push controller
```

Comment:
Can this documentation describe what a `MetricsProcessor` is useful for instead of (or in addition to) where it is useful?  Also, on an interface I'd avoid being too specific about where it is useful as this will be out-of-date once another controller type other than `push` can use a `MetricsProcessor`.